packages/legacy/src/rest-3/errors.ts

```diff
@@ -0,0 +1,7 @@
+export class NotImplementedError extends Error {
+  /* istanbul ignore next */
+  message =
+    process.env.NODE_ENV === 'production'
+      ? /* istanbul ignore next */ 'Not Implemented'
+      : 'Not Implemented. You must provide an override for this method.';
```

Comment:
is there a point in changing this message in prod? only the site devs are going to be looking at it

Reply:
Smaller bundle. React has a better solution where they give an id you can go to a website for when you get errors in prod mode

Reply:
seems like a lot of added complexity to save 46 gzippable characters

Reply:
as the branch has no interactions with other state or branches it adds a complexity of size 1.

Reply:
i'm worried about the human complexity, not the algorithmic complexity (which rarely matters in practice)

Reply:
algorithmic complexity is things like runtime and memory, which relate to the cost of running an algorithm in terms of computational resources with relation to the input size. The only control flow that can affect these things are variable allocations, and loops (recursion is equivalent to loops). Adding branches cannot affect these runtime conditions by themselves.

I don't know what you mean by 'human complexity' but I am speaking to the understandability, testability, provability of the code. If one were to prove the behavior of an algorithm, the complexity I am describing would be the number of distinct cases that would have to be enumerated. Additionally even if some cases are technically equivalent, if it is not obvious that they are, that too must be proven resulting in those cases being added to complexity.

How does this manifest? Nesting conditionals where the variables do not have exclusive state is one way; then if those states can affect other parts of the control flow (by function calls) one must multiply all those possibilities. That's where things like two way data binding become intractable because while yes, they often result in infinite runtime loops they also result in infinite cases of control flow.

The affect of this conditional is isolated. Its only side effect is a console log - which happens in both cases. It has no return value. Without reading any code one can see that there is no way for it to affect the behavior of any other piece. Therefore this conditional provably multiplies by nothing.